### DISCUSSION TITLE
vcpkg port request

### DISCUSSION THREAD

**Discussion by white-echidna** [vsg-dev/VulkanSceneGraph#674] [General]
I would like to use VSG in my project. Is it ready for distribution using vcpkg?

**Comment by white-echidna** [comment]
can you please leave the vcpkg port request for VSM like you did for OSM?

**Comment by robertosfield** [comment]
What would VSM and OSM be?  Do you mean VSG and OSG?

I haven't personally be involved with vspkg support for the OSG side, and can't provide guidance on how to go about vcpkg support.  Perhaps other members of the VulkanSceneGraph community will be able to provide assistance.

As a general note, we've tried to make the VSG and associated libraries easy to build and would encourage users to get comfortable with building and installing themselves as this provide greater flexibility and control.

I will move this discussion from Issue to the Discussion pages as this would be a better venue for further discussion about the topic.

**Reply by white-echidna** [reply]
I understand. But add_subdirectory or FetchContent are hard dependencies needed for the libraries source code (my main dev. env. is Windows, cmake searching dependencies in their isn't good). So, vcpkg just comfortable dependencies provider. May I try to add simple preview vcpkg support in vsg until people who know it better do it?
> What would VSM and OSM be? Do you mean VSG and OSG?
yes, ofcourse. sorry)
> I will move this discussion from Issue to the Discussion...
It's your world

**Comment by white-echidna** [comment]
ok, I created [new port request](https://github.com/microsoft/vcpkg/issues/28529)

**Reply by robertosfield** [reply]
Thanks.  

I don't know what the process is within the vkpg community is, but I would have thought help from members of the VulkanSceneGraph community might be required.

For application developers they will typically want to use the other libraries associated with the VulkanSceneGraph like vsgXchange.  For instance @repagh recently created binary rpm & deb packages, mention on thread #671, The list of packages can be found at:

https://build.opensuse.org/project/show/home:repabuild:vulkanscenegraph

Another avenue to look at would be [conan](https://conan.io/), something that @psinf discusses in the thread #590.
